zerver/lib/actions.py

```diff
@@ -5017,11 +5017,11 @@ def gather_subscriptions_helper(user_profile: UserProfile,
             stream_dict['is_announcement_only'] = \
                 stream['stream_post_policy'] == Stream.STREAM_POST_POLICY_ADMINS
 
-            if is_public or user_profile.is_realm_admin:
-                subscribers = subscriber_map[stream["id"]]
-                if subscribers is not None:
-                    stream_dict['subscribers'] = subscribers
+            subscribers = subscriber_map[stream["id"]]
```

Comment:
This is already inside of a `if is_public or user_profile.is_realm_admin:` condition, line 5003, so I think it can be removed. I ran the `test_subs` test after this removal and it was fine.